src/rest-api.js

```diff
@@ -47,8 +52,29 @@ function perform(config = {}, method = "get", path, params = {}) {
 
     query
     .on("success", actions.resolve)
-    .on("error", actions.reject)
-    .on("fail", actions.reject);
+    .on("error", actions.reject);
+
+    if (method === "get") {
```

Comment:
hm, yes only get requests have 503 handling with retry now.
If I add `options.retry` we will end up basically with such use cases:
```
client.get("/test") -> will be retried on 503 after 100 ms
client.get("/test", { retry: 500 }) -> will be retried on 503 after 500ms

client.post("/test") -> won't be retried
client.post("/test", { retry: 500 }) -> will be retried on 503 after 500ms

client.get("/test", { timout: 1000, retry: 500 }) -> both 503 and timeout errors will be handled, same for all http verbs